### PR TITLE
Add tests for existing JavaScript code

### PR DESCRIPTION
**Pull request by alex-ju**
In the past few weeks we added quite a bunch of JavaScript code to get features out in the wild and test them during user research and so we sinned by not writing unit tests and rely only on functional tests. This PR is trying to address this issue by adding a set of tests for the existing JavaScript code with the hope to add more tests that would cover the server requests.

Run tests locally with `rake jasmine`

jasmine-2

[Trello card](https://trello.com/c/0AXSQGEx)

### CODE REVIEW

File: spec/javascripts/components/error-alert-spec.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  })
+
+  afterEach(function () {
+    element.remove()
+    element = undefined
+  })
+
+  it('should be focused', function () {
+    module.focus()
+    expect(element[0]).toBeFocused()
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply spec/javascripts/components/error-alert-spec.js]: Yes, it should. But we want to ensure the state not the style.

File: spec/javascripts/components/markdown-editor-spec.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    element = undefined
+  })
+
+  describe('when initialised', function () {
+    beforeEach(function () {
+      module.init()
+    })
+
+    it('should display the editor head', function () {
+      var head = $('.js-markdown-editor__head')
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply spec/javascripts/components/markdown-editor-spec.js]: Not my choice of naming. Would _should display the editor header_ be more decent?

File: spec/javascripts/components/error-alert-spec.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (benthorner) [reply spec/javascripts/components/error-alert-spec.js]: 👍 Just wanted to check you'd seen it.

File: spec/javascripts/components/markdown-editor-spec.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (benthorner) [reply spec/javascripts/components/markdown-editor-spec.js]: Ah, I see. Using 'header' would make more sense, although I wonder if this is actually covered by the clicking buttons specs below? Would they pass if this spec failed?

File: spec/javascripts/helpers/jasmine_examples/SpecHelper.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+beforeEach(function () {
+  jasmine.addMatchers({
+    toBeFocused: function () {
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply spec/javascripts/helpers/jasmine_examples/SpecHelper.js]: We can use [jasmine-jquery](https://github.com/velesin/jasmine-jquery) to be consistent with the govuk_publishing_components as we need to be able to move these components with tests upstream.

**Comment by alex-ju** [comment]
@benthorner addressed your comments (including redundant tests and nesting). Regarding the matchers we don't have much flexibility as we need to be able to move them in the shared library.

File: spec/javascripts/helpers/jasmine_examples/SpecHelper.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (alex-ju) [reply spec/javascripts/helpers/jasmine_examples/SpecHelper.js]: Sounds good. Will do that.

**Pull request merged**